app/controllers/collections_controller.rb

```diff
@@ -138,17 +142,21 @@ def get_revisions(concept_id, revision_id)
     revisions
   end
 
-  def set_collection_link(concept_id)
-    # collection_link used for downloading XML
-    base_url = 'http://localhost:3003'
-    if Rails.env.sit?
-      base_url = 'https://cmr.sit.earthdata.nasa.gov/search'
-    elsif Rails.env.uat?
-      base_url = 'https://cmr.uat.earthdata.nasa.gov/search'
-    elsif Rails.env.production?
-      base_url = 'https://cmr.earthdata.nasa.gov/search'
-    end
-    @collection_link = "#{base_url}/concepts/#{concept_id}"
+  def set_collection_download
+    @concept_id = params[:id]
+    @collection_format = params[:format]
+    revision_id = params[:revision_id]
+
+    collection_response = cmr_client.get_collection_concept_for_download(@concept_id, @collection_format, token, revision_id)
+
+    @collection_download = if collection_response.success?
+                             collection_response.body
+                           else
+                             Rails.logger.error("Error retrieving concept for Collection #{@concept_id}, revision #{revision_id} in #{@collection_format} format for user #{current_user.urs_uid}")
+
+                             error_message = "There was an error retrieving the collection metadata for Collection #{@concept_id}, revision #{revision_id} in #{@collection_format} format that you requested."
+                             "<?xml version='1.0' encoding='UTF-8'?><error>#{error_message}</error>"
```

Comment:
I'm 50/50 on supplying the error in the file. 

1. If the file isn't xml (when the native format is umm-json for instance) this would be odd.
2. Is there a better way to display this? A redirect or something? We may want to punt on trying to show the error given that

    - this wasn't something we did before this change
    - variable json download doesnt do this
    - we have an upcoming ticket to display appropriate error pages
    - it seems unlikely to throw an error given that were looking at the show page when this link is displayed.

Reply:
1. I agree but maybe this part could be handled 

1. I can certainly not provide the error in the file, but I'm not sure there's a better way to display it:
  - `flash.now` didn't work when I tried so I'm not sure how any flash type message might work
  - A redirect won't work because the link isn't taking the user anywhere it is initiating a download with an empty tab opening at most

I agree that the show page was being viewed so an error is unlikely, so I can remove the error in the file here. But -- if there is an error should we provide an empty file or not allow a download?